consensus/src/lib.rs

```diff
@@ -99,40 +99,55 @@ impl<'a> ConsensusRef<'a> {
         let block_index = self.db_tx.get_block_index(block_id).map_err(BlockError::from)?;
         match block_index {
             Some(block_index) => {
-                let prev_block =
+                let prev_block_id =
                     block_index.get_prev_block_id().as_ref().ok_or(BlockError::NotFound)?;
-                Ok(self.db_tx.get_block_index(prev_block)?.ok_or(BlockError::NotFound)?)
+                Ok(self.db_tx.get_block_index(prev_block_id)?.ok_or(BlockError::NotFound)?)
             }
             None => Err(BlockError::NotFound),
         }
     }
 
-    fn get_block_from_index(&self, block_index: &BlockIndex) -> Result<Option<Block>, BlockError> {
-        Ok(self.db_tx.get_block(block_index.get_block_id().clone())?)
+    // Get indexes for a new longest chain
+    fn get_new_chain(
```

Comment:
Something seems off to me here. Since we always insert into position zero this means we will always return a vector of length at most one. Am I missing something?

Reply:
No! When we insert a new item, we do not remove the previous item. As a result, we have here the `Vec` with reverse order.  Have a look [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1a29b3dc27a2c128f31aa7a848bc69e8).

Reply:
Gotcha! Thanks.